packages/shared/src/components/post/common/PostBackButton.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { ReactElement } from 'react';
+import { useRouter } from 'next/router';
+import { ButtonVariant } from '../../buttons/common';
+import ArrowIcon from '../../icons/Arrow';
+import { Button } from '../../buttons/ButtonV2';
+import { FeedLayout } from '../../../lib/featureValues';
+import { useFeature } from '../../GrowthBookProvider';
+import { feature } from '../../../lib/featureManagement';
+
+interface PostBackButtonProps {
+  link?: string;
+}
+
+export function PostBackButton({ link }: PostBackButtonProps): ReactElement {
+  const router = useRouter();
+  const layout = useFeature(feature.feedLayout);
+  const canGoBack = !!globalThis?.window?.history?.length;
+  const onClick = () => router.push(link);
+
+  if (layout === FeedLayout.Control || (!canGoBack && !link)) {
```

Comment:
Why do we check for control rather then not being on V1?

Reply:
These parts are regardless of which feed you are in since this is the post page which is never in a feed.

Reply:
To return early and render null.